ephemeris_tools_unit_tests/ephem_tools_unit_tests.py

```diff
@@ -0,0 +1,702 @@
+    else:
+        assert server == 'other', 'not an accepted value for server'
+        selected_server = str(input('type the URL prefix '
+                                    'for the server you wish to use: '))
+        logging.info(f'Alternative server chosen: {selected_server}')
+        
+    golden_path = golden_location[0]
+    
+    # Having a file_type variable ensures that the golden copy counterpart will
+    # be cleaned with the correct cleaning code. Since the golden copy will
+    # always be the same file type as the test copy, they will use the same
+    # cleaning code.
+    file_type = None
+    tool_type = None
+
+    logging.info('Beginning test file vs. golden copy comparison')
+    
+    for ephemeris_type in ephem:
+        if ephemeris_type == 'test':
+            suffix = '&ephem=+-1+TEST&sc_trajectory=+-1+TEST'
+        else:
+            assert ephemeris_type == 'current'
+            suffix = ''
+
+        for test in tests_final:
+            if 'viewer3' in test:
+                tool_type = 'Planet Viewer test'
+                if 'output=HTML' in test:
+                    index = list(viewer_urls).index(test) + 1
+                else:
+                    assert'output=PS' in test
+                    index = viewer_indices_ps.index(test) + 1
+            
+            if 'tracker3' in test:
+                tool_type = 'Moon Tracker test'
+                if 'output=HTML' in test:
+                    index = list(tracker_urls).index(test) + 1
+                elif'output=PS' in test:
+                    index = tracker_indices_ps.index(test) + 1
+                else:
+                    assert 'output=TAB' in test
+                    index = tracker_indices_tab.index(test) + 1
+            
+            if 'ephem3' in test:
+                tool_type = 'Ephemeris Generator test'
+                if 'output=HTML' in test:
+                    index = list(ephemeris_urls).index(test) + 1
+                else:
+                    assert'output=TAB' in test
+                    index = ephemeris_indices_tab.index(test) + 1
+            name = str(uuid.uuid5(uuid.NAMESPACE_URL, test))
+            test = selected_server + test + suffix
+            content = requests.get(test, verify=False).content
+            content = content.decode('utf8')
+            if 'output=HTML' in test:
+                test_version = html_file_cleaner(ephem, content)
+                file_type = 'html'
+            elif 'output=PS' in test:
+                test_version = ps_file_cleaner(ephem, content)
+                file_type = 'ps'
+            else:
+                assert 'output=TAB' in test
+                file_type = 'tab'
+        
+            try:
+                with open(os.path.join(golden_path, name), 'r') as file_to_read:
+                    golden_file = file_to_read.read()
+            except FileNotFoundError:
+                logging.warning(f'Filename {name} not found within directory')
```

Comment:
If the file is not found, then `golden_file` is never set, which means the following statements will crash.